sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -836,6 +837,61 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(cast("", BooleanType), null)
   }
 
+  test("cast integer types to binary") {
+    val bos = new ByteArrayOutputStream()
+    val dos = new DataOutputStream(bos)
+
+    def withDos(f: DataOutputStream => Unit): UTF8String = {
+      f(dos)
+      dos.flush()
+      val res = UTF8String.fromBytes(bos.toByteArray)
+      bos.reset()
+      res
+    }
+
+    def checkBinaryCast[T](in: T): Unit = in match {
+      case b: Byte =>
+        checkEvaluation(cast(cast(in, BinaryType), StringType), withDos(_.writeByte(b)))
+      case s: Short =>
+        checkEvaluation(cast(cast(in, BinaryType), StringType), withDos(_.writeShort(s)))
+      case i: Int =>
+        checkEvaluation(cast(cast(in, BinaryType), StringType), withDos(_.writeInt(i)))
+      case l: Long =>
+        checkEvaluation(cast(cast(in, BinaryType), StringType), withDos(_.writeLong(l)))
+    }
+
+    // Byte
+    checkBinaryCast(0.toByte)
+    checkBinaryCast(1.toByte)
+    checkBinaryCast(-1.toByte)
+    checkBinaryCast(Byte.MaxValue)
+    checkBinaryCast(Byte.MinValue)
+
+    // Short
+    checkBinaryCast(0.toShort)
+    checkBinaryCast(1.toShort)
+    checkBinaryCast(-1.toShort)
+    checkBinaryCast(Short.MaxValue)
+    checkBinaryCast(Short.MinValue)
+
+    // Int
+    checkBinaryCast(0)
+    checkBinaryCast(1)
+    checkBinaryCast(-1)
+    checkBinaryCast(Int.MaxValue)
+    checkBinaryCast(Int.MinValue)
+
+    // Int
+    checkBinaryCast(0L)
+    checkBinaryCast(1L)
+    checkBinaryCast(-1L)
+    checkBinaryCast(Long.MaxValue)
+    checkBinaryCast(Long.MinValue)
+
+    dos.close()
+    bos.close()
+  }
+
```

Comment:
I think we'd be better to add new tests in cast.sql: https://github.com/apache/spark/blob/cad475dcc9376557f882859856286e858002389a/sql/core/src/test/resources/sql-tests/inputs/cast.sql#L45

Reply:
I added acceptance tests for integral values. To validate the output, I created a hex string. The pretty format for binary values seems to be their corresponding ASCII values.

Reply:
I removed the tests from `CastSuite`